### PR TITLE
Fix topk accuracies

### PR DESCRIPTION
**Pull request by antonschafer**
- Fixes bug where padding was counted in the number of correct topk predictions, leading to potentially inflated accuracy scores when the eval source was using padding
- Unifies and simplifies the way iteration over the eval data source is done. Reduces memory footprint. This also fixes an off by one bug where the evaluation runs for an extra step (see [here](https://github.com/languini-kitchen/languini-kitchen/blob/main/languini/train_lib/lm_trainer.py#L94))

### CODE REVIEW

File: languini/train_lib/lm_trainer.py [hunk 1]
```diff
@@ -128,14 +123,12 @@ def evaluation(config, model, state, data_source, max_steps, last_n=-1, print_pr
                 check(all_losses, (bsz * last_n,))
 
                 # mask losses that are padded (unlike training, evaluation can result in batches with padded batches)
-                if is_padded:
-                    all_losses = all_losses.masked_fill(batch_x.reshape(-1) == 0, 0.0)
-                    token_count = torch.sum(batch_x != 0)
-                else:
-                    token_count = torch.tensor([batch_x.numel()], device=all_losses.device)
+                is_padding = batch_x.reshape(-1) == 0 # (bsz * last_n,)
```
Comment (antonschafer) [inline languini/train_lib/lm_trainer.py:126]: Doing this always now, no matter if `is_padded` or not. Simplifies code and shouldn't lead to any noticeable slowdown. If you prefer, will reintroduce distinction and introduce a special is_padding=None case for `total_correct`
